gyft.py

```diff
@@ -126,42 +150,45 @@ def merge_slots(in_dict):
     for time in timetable_dict[day]:
         timetable_dict[day][time].append(" ")
         course_code = timetable_dict[day][time][0]
-        course_dept = course_code[:2]
-        if course_dept not in courses.keys():
-            courses[course_dept] = {}
-        if course_code not in courses[course_dept].keys():
-            courses[course_dept][course_code] = ''
-
-# scraping course names deptwise
-for dept in courses.keys():
-    DEPT_URL = COURSES_URL.format(dept)
-    r = s.get(DEPT_URL, headers=headers)
-    soup = bs(r.text, 'html.parser')
-    parentTable = soup.find('table', {'id': 'disptab'})
-
-    rows = parentTable.find_all('tr')
+        course_dept = course_code[:2] 
+        try:
+            dept_code = dept_codes[course_dept]
+        except:
+            dept_code = input("Enter the 2 letter dept code for {}: ".format(course_dept))
+            print()
+            dept_codes[course_dept] = dept_code
+            updated_codes = True
 
-    for row in rows[1:]:
-        if 'bgcolor' in row.attrs:
-            continue 
-        cells = row.find_all('td')  
-        course_code = cells[0].text.strip()
-        course_name = cells[1].text.strip()
+        if dept_code not in courses.keys():
+            courses[dept_code] = {}
+        if course_code not in courses[dept_code].keys():
+            courses[dept_code][course_code] = ''
 
-        if course_code in courses[dept]:
-            courses[dept][course_code] = course_name
-            logging.info(" {} - {}".format(course_code, course_name))
 
+# scraping course names deptwise
+for dept in courses.keys():
+    try: 
+        scrape_courses(courses,dept)
+    except:
+        print()
+        logging.error(" Error while scraping course names for course letter code {}".format(dept))
+        logging.info(" You can manually add the course names while executing generate_ics.py")
+        print()
 
 # add course code to dict
 for day in timetable_dict.keys():
     for time in timetable_dict[day]:
         course_code = timetable_dict[day][time][0]
-        course_dept = course_code[:2]
-        timetable_dict[day][time][3] = courses[course_dept][course_code]
+        dept_code = dept_codes[course_code[:2]]
+        timetable_dict[day][time][3] = courses[dept_code][course_code]
 
 
 with open('data.txt', 'w') as outfile:
     json.dump(timetable_dict, outfile, indent = 4, ensure_ascii=False)
 
-print ("\nTimetable saved to data.txt file. Be sure to edit this file to have desired names of subjects rather than subject codes.\n")
+if updated_codes:
+    with open('dept_codes.json', 'w') as outfile:
+        json.dump(dept_codes, outfile, indent = 4, ensure_ascii=False)
+    print("\nThe dept_codes.json file has been updated with the new dept codes. Please send a PR to the repo with the updated file.")
```

Comment:
```suggestion
    print("\nThe dept_codes.json file has been updated. Please send a PR to the repository for the same.")
```

Reply:
Seems like this issue is because AI courses started after `what-slot` was made lol, so it wasn't in the `deps.txt`, and hence it wasn't scraped. 

Removing the cases with no mismatch can be done, but I thought for sake of completeness, this is better, as for all depts, the dict can be referred to, rather than some referring to dict and some not. 

And I don't think it is that much of an issue because there are only ~50 codes, compared to >500 subjects in `subjects.json` and chances of more codes being introduced are less.

There is one alternative (diferent source for scraping) I had tried even in `gyfe`, but didn't work back then, will try it again. For now though, I think this works well enough.

Reply:
- Still, it will just lead to errors with cases which would have been solved easily

Why to even do that ¯\_(ツ)_/¯